avstack/modules/perception/fov_estimator.py

```diff
@@ -152,6 +156,24 @@ def _estimate_fov_from_polar_lidar(
     ) -> "Polygon":
         """To be implemented in subclass"""
         raise NotImplementedError
+    
+    def _eliminate_isolated_pts(self, pc_bev, m_away, num_pts):
+        ptMap = defaultdict(int)
+        usable_pts = []
+        for p1 in pc_bev.data.x:
```

Comment:
You don't actually need the loops to do the distance. You can do broadcasting in numpy to make this much faster. See e.g., https://sparrow.dev/pairwise-distance-in-numpy/

Also note that the distance is commutative, meaning dist(a, b) = dist(b, a) so even if you did need loops, you could make the first loop `for i in range(0, len(pc_bev.data.x))` and the second loop `for j in range(i, len(pc_bev.data.x))` which would cut out half of the evaluations.